src/web_or_worker_shell_read.js

```diff
@@ -20,7 +20,19 @@
       return new Uint8Array(/** @type{!ArrayBuffer} */(xhr.response));
     };
   }
-
+#if USE_FETCH
+  readAsync = (url, onload, onerror) => {
+    fetch(url)
+    .then(response => {
+      if(response.ok) {
+        return response.arrayBuffer();
+      }
+      throw new Error(response.statusText + ' : ' + response.url);
```

Comment:
Should this not call `onerror` instead?

Reply:
Let me check the make sure the error has the correct stack if its not actually thrown, and then I can change it.

Reply:
I remembered why I used this pattern as I was editing: The `catch()` at the end of the function will catch errors coming from the `fetch()` itself, but `fetch()` won't throw errors by itself if the request itself fails, i.e. in a 404 or 500 error. For that we have to throw our own error. When we do that, we skip the `then()` on line 32 and head right to `catch()`.

Reply:
How about `return new Promise.reject(..)`

Then I think maybe you can do `.then(onload, onerror)` without the extra catch?   Not sure which is more idomatic.